### ISSUE TITLE
Failed to activate the default-language package

### ISSUE THREAD

**Issue by fstani** [Fankserver/atom-default-language#6]
[Enter steps to reproduce:]

1. Starting atom normally

**Atom**: 1.23.1 x64
**Electron**: 1.6.15
**OS**: Mac OS X 10.13.1
**Thrown From**: [default-language](https://github.com/Fankserver/atom-default-language) package 0.3.0

### Stack Trace

Failed to activate the default-language package

```
At Cannot read property 'toLowerCase' of undefined

TypeError: Cannot read property 'toLowerCase' of undefined
    at Object.checkGrammar (/packages/default-language/lib/index.coffee:54:28)
    at /packages/default-language/lib/index.coffee:21:67
    at PaneContainer.observePaneItems (/Applications/Atom.app/Contents/Resources/app/src/pane-container.js:107:51)
    at WorkspaceCenter.observePaneItems (/Applications/Atom.app/Contents/Resources/app/src/workspace-center.js:71:65)
    at /Applications/Atom.app/Contents/Resources/app/src/workspace.js:745:32
    at Array.map (native)
    at Workspace.observePaneItems (/Applications/Atom.app/Contents/Resources/app/src/workspace.js:744:116)
    at Object.activate (/packages/default-language/lib/index.coffee:21:39)
    at Package.module.exports.Package.activateNow (/Applications/Atom.app/Contents/Resources/app/src/package.js:255:25)
    at /Applications/Atom.app/Contents/Resources/app/src/package.js:227:38
    at Package.module.exports.Package.measure (/Applications/Atom.app/Contents/Resources/app/src/package.js:99:21)
    at /Applications/Atom.app/Contents/Resources/app/src/package.js:220:32
    at Package.module.exports.Package.activate (/Applications/Atom.app/Contents/Resources/app/src/package.js:217:40)
    at PackageManager.activatePackage (/Applications/Atom.app/Contents/Resources/app/src/package-manager.js:695:42)
    at config.transactAsync (/Applications/Atom.app/Contents/Resources/app/src/package-manager.js:670:36)
    at Config.module.exports.Config.transactAsync (/Applications/Atom.app/Contents/Resources/app/src/config.js:367:24)
    at PackageManager.activatePackages (/Applications/Atom.app/Contents/Resources/app/src/package-manager.js:668:23)
    at PackageManager.activate (/Applications/Atom.app/Contents/Resources/app/src/package-manager.js:647:50)
    at /Applications/Atom.app/Contents/Resources/app/src/atom-environment.js:865:36
```

### Commands

```
```

### Non-Core Packages

```
atom-beautify 0.30.9 
colorblind-syntax 2.0.0 
colorblind-wombat-dark-syntax 0.6.0 
default-language 0.3.0 
des-dark-syntax 0.2.0 
language-perl-html-template 2.1.0 
minimap 4.29.7 
monokai-blackboard 0.3.0 
multi-cursor 2.1.5 
multiline-select 1.8.2 
nucleus-dark-ui 0.12.3 
revert-buffer 0.6.0 
spacegray-dark-neue-ui 1.7.1 
tabs-to-spaces 1.0.3 
```

**Comment by joelcarvalhais** [comment]
Same here!
Any update on this? or any other solution?

**Comment by fank** [comment]
I currently do not have time to take a look in this sorry.

**Comment by johntreankler** [comment]
I'm using this work-around: **Edit** > **Select Grammar** > (select from list).

**Comment by fank** [comment]
There is a shortcut for it, it may help

**Comment by joelcarvalhais** [comment]
thank you :) im using the shortcut on the bottom right of the page

**Comment by johntreankler** [comment]
Oh, nice! That's a little quicker than the **Edit** menu. 😸

**Comment by ColoradoWebGuy** [comment]
@Fank 

I'm also using the work-around: **Edit** > **Select Grammar** > (select from list), or by simply selecting it through the shortcut on the bottom right corner of the IDE (Atom).

It still isn't as cool as having the default language set though. It's annoying to have to "Select Grammar" every time a file is opened. I loved this plugin, but since it's now deprecated, are there any other ways for my system to open the default language configuration of my choosing?

**Comment by burneyhoel** [comment]
It looks like the error is coming from a language package that does not assign 'scopeName'. @fstani  example language-perl-html-template and the core package language-text do not assign 'scopeName'. I hope this helps.

**Comment by joelcarvalhais** [comment]
after the Atom update the issue seems to be solved

**Comment by burneyhoel** [comment]
@joelcarvalhais I am on 1.23.2 and still receiving this error, but only for certain languages.

**Comment by joelcarvalhais** [comment]
@theexec im on that version too. For sql and shell is working well. not sure about the rest =/

**Comment by burneyhoel** [comment]
While trying a few things to find a solution, I uninstalled this package and reinstalled while on the latest atom 1.23.2 and the issue appears to have been resolved.

**Comment by ColoradoWebGuy** [comment]
@theexec @joelcarvalhais - Thank you!!

You're both right! After I _updated Atom_ AND _uninstalled this package and reinstalled_. . . then restarted Atom, it works again. Yay!!

**Comment by johntreankler** [comment]
Happy to see it's working for y'all, but: I updated Atom to 1.23.3, uninstalled **default-language** package, quit Atom, opened Atom, and when I re-install I *still* get "Failed to Activate" due to "Cannot read property 'toLowerCase' of undefined" 😿 so YMMV.

**Comment by burneyhoel** [comment]
@johntreankler - what language packages do you have active? 

I just tested a few other things and I am still getting this error under the right conditions. It seems that if i have a tab open that requires the default language package to run as the program is starting and I have a language enabled that does not assign 'scopeName' then it fails.

**Comment by johntreankler** [comment]
@theexec Only language package I have installed is **language-pick-basic**. I'll try some other things based on your feedback & report back here if it succeeds. Thanks!

**Comment by theexec** [comment]
@johntreankler Try disabling the Core package language-text.

![image](https://user-images.githubusercontent.com/13513687/34788213-d7de487e-f607-11e7-85e4-e979ae8bb237.png)

**Comment by burneyhoel** [comment]
@johntreankler I stand corrected, the plain text package is not the issue. I think that if when you open atom if all your tabs have a valid language already selected the package will initialize and be fine after that. I also am not sure if i really know what I am talking about at this point. I think @Fank will need to address this when he has time.

**Comment by johntreankler** [comment]
Huzzah! That did it. Thanks so much, @theexec 😄

**Comment by fstani** [comment]
Yes, it seems to be something in the initialization / execution time.

I often have to restart atom because I use a SSHFS setup for some stuff, and when re-loading it happens more often, @theexec must be on something considering the tabs being opened.

I use these package because we have files with no extension on the project, so I wanted something to default them to a particular language.

I can debug the issue, we can probably avoid the runtime error by checking the type of object first, but i don't know if that will fix it.

I will give it a shot in changing the coffee code

**Comment by dgleich** [comment]
It seems like this should be a pretty simple fix... 

On line 54 of `lib/index.coffee`, change it to...
~~~~
if grammar.name && grammar.name.toLowerCase().replace(/\ /g, '') is newGrammar.toLowerCase().replace(/\ /g, '')
~~~~

This is what the grammar-list-view uses to pick valid grammars to list... (https://github.com/atom/grammar-selector/blob/master/lib/grammar-list-view.js, see line 72-74)

I will try and submit this as a simple pull request, but maybe one of the developers can just insert it...

**Comment by fank** [comment]
Thanks to @dgleich and published 0.3.1
